hyp3_autorift/vend/testautoRIFT.py

```diff
@@ -199,13 +199,19 @@ def runAutorift(I1, I2, xGrid, yGrid, Dx0, Dy0, SRx0, SRy0, CSMINx0, CSMINy0, CS
     #        We should not mask based on zero values in the L7 images as this percolates into SearchLimit{X,Y}
     #        and prevents autoRIFT from looking at large parts of the images, but untangling the logic here
     #        has proved too difficult, so lets just turn it off if `wallis_fill` preprocessing is going to be used.
+    #        However, we do have the image zero_mask already, so we can use that to create the output product noDataMask
     # generate the nodata mask where offset searching will be skipped based on 1) imported nodata mask and/or 2) zero values in the image
     if 'wallis_fill' not in preprocessing_methods:
         for ii in range(obj.xGrid.shape[0]):
             for jj in range(obj.xGrid.shape[1]):
                 if (obj.yGrid[ii,jj] != nodata)&(obj.xGrid[ii,jj] != nodata):
                     if (I1[obj.yGrid[ii,jj]-1,obj.xGrid[ii,jj]-1]==0)|(I2[obj.yGrid[ii,jj]-1,obj.xGrid[ii,jj]-1]==0):
                         noDataMask[ii,jj] = True
+    elif zero_mask is not None:
+        for ii in range(obj.xGrid.shape[0]):
+            for jj in range(obj.xGrid.shape[1]):
+                if (obj.yGrid[ii, jj] != nodata) & (obj.xGrid[ii, jj] != nodata):
```

Comment:
Just want to confirm that the use of the bitwise `&` operator instead of the boolean `and` is intentional? (I don't really understand the logic behind how the `noDataMask` is being computed here.)

Reply:
I guess it shouldn't actually make a difference, but it just confused me as to whether or not this is a boolean operation.

Reply:
@jtherrmann Yes, that will end up operating on two boolean values so `&` and `and` are functionally equivalent. I think `and` is more idiomatic here, but since the `&` is used in the previous block and it's *their* code, I'm keeping the style consistent by using `&` here.

Reply:
Makes sense!